### ISSUE TITLE
Establish a list of ESLint rules that are relevant for component authoring

### ISSUE THREAD

**Issue by Princesseuh** [Princesseuh/component-template#1]
Currently, the only special ESLint rule is a `error` on `console.log` (but not `console.error` and others), the reason for that being that they'll leak in the user's console when using the component which, in my opinion is undesirable

In this issue, I'd love to figure out if there's any other settings we should have for component authoring. This is part of a bigger issue of establish a "code" to respect when authoring Astro components that are meant for distribution

Basically, is there any code that a developer should avoid putting in their project because it would cause issue when users use it and if so, what ESLint rules can we add to catch this?
